src/http/validations/device.js

```diff
@@ -13,6 +14,19 @@ const getSingleDevice = (req, res, next) => {
   return next();
 };
 
-export default {
-  getSingleDevice,
+const updateDevice = (req, res, next) => {
+  const { deviceId } = req.params;
+
+  const fieldErrors = new FieldErrors();
+
+  if (!db.validResourceId(deviceId)) fieldErrors.addError('deviceId', 'deviceId should be a ');
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is a required string');
+  if (req.body.constructor === Object && Object.keys(req.body).length === 0) fieldErrors.addError('request', 'Request body can\'t be empty');
+
```

Comment:
We still need validation for the `label` field, however, the validation should only execute if the label is provided.

Reply:
fixed